src/model/bpmn/internal/shape/utils.ts

```diff
@@ -45,6 +45,18 @@ export class ShapeUtil {
     return ShapeBpmnElementKind.EVENT_START === kind;
   }
 
+  static isCatchEvent(kind: ShapeBpmnElementKind): kind is ShapeBpmnElementKind.EVENT_INTERMEDIATE_CATCH | ShapeBpmnElementKind.EVENT_BOUNDARY | ShapeBpmnElementKind.EVENT_START {
+    return ShapeBpmnElementKind.EVENT_INTERMEDIATE_CATCH === kind || ShapeBpmnElementKind.EVENT_BOUNDARY === kind || ShapeBpmnElementKind.EVENT_START === kind;
+  }
+
+  static isIntermediateCatchEvent(kind: ShapeBpmnElementKind): kind is ShapeBpmnElementKind.EVENT_INTERMEDIATE_CATCH {
```

Comment:
**question:** I don't understand the need of the "is ..." in the signature. Why not simply use a boolean as in other methods?
This feature is not used where this method is called.

Same remark for the  `isIntermediateThrowEvent` method.

For `isCatchEvent`, the feature is used but it could probably be avoided by relaxing the type of the `elementKind` parameter in `ProcessConverter.buildShapeBpmnCatchEvent` as it is a private method so we know what we are doing.

Reply:
:information\_source: [https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates)  
  
This avoid to cast the kind where we use these functions.  
  
If I replace the returned type by boolean, I have the following error:

![](https://github.com/process-analytics/bpmn-visualization-js/assets/4921914/63277eae-09b2-472d-afff-6d424a70fc5f)

Reply:
Done in #2924